pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -660,3 +676,122 @@ func parseResourceGroupHashFromAnnotation(s *fleetv1beta1.ClusterResourceSnapsho
 	}
 	return v, nil
 }
+
+// TODO: handle multiple resourceSnapshots
+func (r *Reconciler) buildPlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, latestSchedulingPolicySnapshot *fleetv1beta1.ClusterSchedulingPolicySnapshot, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) (*fleetv1beta1.ClusterResourcePlacementStatus, error) {
+	status := &fleetv1beta1.ClusterResourcePlacementStatus{
+		SelectedResources: make([]fleetv1beta1.ResourceIdentifier, 0, len(latestResourceSnapshot.Spec.SelectedResources)),
+		PlacementStatuses: make([]fleetv1beta1.ResourcePlacementStatus, 0, 20), // pre-allocate with a reasonable capacity
+		Conditions:        []metav1.Condition{},
+	}
+	// TODO query the sub resourceSnapshots by using latestResourceSnapshot and then update the status.
+	for i := range latestResourceSnapshot.Spec.SelectedResources {
+		obj, err := r.decodeResourceContent(&latestResourceSnapshot.Spec.SelectedResources[i])
+		if err != nil {
+			klog.ErrorS(err, "Failed to encode resourceContent of clusterResourceSnapshot", "clusterResourcePlacement", klog.KObj(crp), "clusterResourceSnapshot", klog.KObj(latestResourceSnapshot))
+			return nil, controller.NewUnexpectedBehaviorError(err)
+		}
+		ri := fleetv1beta1.ResourceIdentifier{
+			Group:     obj.GroupVersionKind().Group,
+			Version:   obj.GroupVersionKind().Version,
+			Kind:      obj.GroupVersionKind().Kind,
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		}
+		status.SelectedResources = append(status.SelectedResources, ri)
+	}
+	scheduledCondition := buildScheduledCondition(crp, latestSchedulingPolicySnapshot)
+	meta.SetStatusCondition(&status.Conditions, scheduledCondition)
+
+	if scheduledCondition.Status == metav1.ConditionUnknown {
```

Comment:
what if it's failed?

Reply:
added a comment in the code to explain why it's unknown only.